### ISSUE TITLE
Material UI icons breaks when migrating to Vite

### ISSUE THREAD

**Issue by tkchan212** [vitejs/vite-plugin-react#74]
### Describe the bug

I had a React project built with create-react-app which I just migrated to Vite. 
Previously the icons where displayed normally, but after migrating to Vite, they become like this
<img width="526" alt="Screenshot 2022-12-30 at 10 21 36 AM" src="https://user-images.githubusercontent.com/77660456/210028146-885c40a5-2b1b-4d2a-98ee-218e54169ace.png">

The way I used material UI icons is like below 
https://mui.com/material-ui/icons/#icon-font-icons
Is there a way to restore the icons?
Any help would be appreciated. Thanks!

### Reproduction

https://stackblitz.com/edit/vitejs-vite-m52qqp

### System Info

```shell
System:
    OS: macOS 12.5
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 1.60 GB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.14.2 - ~/.nvm/versions/node/v16.14.2/bin/node
    Yarn: 1.22.15 - ~/.nvm/versions/node/v16.14.2/bin/yarn
    npm: 8.5.0 - ~/.nvm/versions/node/v16.14.2/bin/npm
  Browsers:
    Chrome: 108.0.5359.124
    Firefox: 108.0.1
    Safari: 15.6
  npmPackages:
    @vitejs/plugin-react: ^3.0.0 => 3.0.0 
    vite: ^4.0.0 => 4.0.3 
Note that this issue exist on Safari/Firefox/Chrome
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite-plugin-react/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite-plugin-react/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitejs/vite-plugin-react/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Vite issue and not a framework-specific issue.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite-plugin-react/discussions) or join our [Discord Chat Server](https://chat.vitejs.dev/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ArnaudBarre** [comment]
From what I understand you try to use the icon font.
From the [documentation](https://mui.com/material-ui/icons/#icon-font-icons) this requires to add this to the head of `index.html`: 
```html
<link
  rel="stylesheet"
  href="https://fonts.googleapis.com/icon?family=Material+Icons"
/>
```

**Comment by tkchan212** [comment]
Dear Arnaud, 
Thanks for the reply. I tried adding the CSS to index.html on the stackblitz sample and it worked immediately, but it didn't change anything on my own project, so I was comparing my own project and the one on stackblitz.

- I noticed that the react and react-dom version used was different, so I updated my react, but nothing seems to change 
- I tried to see what happens if I use a stackblitz sample with TS instead of JS, and there's no issue of broken icons as well.
I don't have any clue right now, so I will switch to iconify first, let's see if there's anyone else with the same issue.
